### PR TITLE
Replace wdio-interface with logUpdate

### LINKED ISSUE #3437: Noise in logs
I went through "Getting started" tutorial https://webdriver.io/docs/gettingstarted.html and I prepared new project.

**Environment:**
 - **WebdriverIO version:** 5.4.7
 - **Mode:** WDIO Testrunner
 - **If WDIO Testrunner, running sync/async:** sync
 - **Node.js version:** 10.15.0
 - **NPM version:** 6.4.1
 - **Installed packages in package.json:**
```
  "devDependencies": {
    "@wdio/cli": "^5.4.7",
    "@wdio/dot-reporter": "^5.4.3",
    "@wdio/junit-reporter": "^5.4.3",
    "@wdio/local-runner": "^5.4.7",
    "@wdio/mocha-framework": "^5.4.6",
    "@wdio/selenium-standalone-service": "^5.2.2",
    "@wdio/sync": "^5.4.6",
    "chromedriver": "^2.45.0",
    "wdio-chromedriver-service": "^5.0.0"
  }
```

I was changing `logLevel` to different values (e.g. `error,` `silent`) in `wdio.conf.js` then I run `./node_modules/.bin/wdio --logLevel silent wdio.conf.js`
Unfortunately I got quite long logs as you can check in details. Is it possible to log only messages suitable for specific log level?

<details>

```
./node_modules/.bin/wdio --logLevel silent wdio.conf.js
2019-01-26T23:12:21.792Z DEBUG wdio-config: @wdio/sync found, running tests synchronous
2019-01-26T23:12:22.202Z DEBUG wdio-cli:run: Run suite with config wdio.conf.js and params { _: [ 'wdio.conf.js' ],
  logLevel: 'silent',
  l: 'silent',
  '$0': 'node_modules/.bin/wdio' }

 RUNNING  0-0 in chrome - /test/specs/google.test.js

Stdout:
2019-01-26T23:12:22.300Z INFO wdio-cli:Launcher: Run onPrepare hook
Starting ChromeDriver 2.45.615355 (d5698f682d8b2742017df6c81e0bd8e6a3063189) on port 9515
Only local connections are allowed.
2019-01-26T23:12:25.226Z INFO wdio-local-runner: Start worker 0-0 with arg: --logLevel,silent,wdio.conf.js

Test Suites:	 0 passed, 1 total (0% completed)
Time:		 🕖  4.87s

 RUNNING  0-0 in chrome - /test/specs/google.test.js

Stdout:
2019-01-26T23:12:22.300Z INFO wdio-cli:Launcher: Run onPrepare hook
Starting ChromeDriver 2.45.615355 (d5698f682d8b2742017df6c81e0bd8e6a3063189) on port 9515
Only local connections are allowed.
2019-01-26T23:12:25.226Z INFO wdio-local-runner: Start worker 0-0 with arg: --logLevel,silent,wdio.conf.js

Test Suites:	 0 passed, 1 total (0% completed)
Time:		 🕚  7.66s

 RUNNING  0-0 in chrome - /test/specs/google.test.js

Stdout:
2019-01-26T23:12:22.300Z INFO wdio-cli:Launcher: Run onPrepare hook
Starting ChromeDriver 2.45.615355 (d5698f682d8b2742017df6c81e0bd8e6a3063189) on port 9515
Only local connections are allowed.
2019-01-26T23:12:25.226Z INFO wdio-local-runner: Start worker 0-0 with arg: --logLevel,silent,wdio.conf.js

Test Suites:	 0 passed, 1 total (0% completed)
Time:		 🕛  7.72s

 RUNNING  0-0 in chrome - /test/specs/google.test.js

Stdout:
2019-01-26T23:12:22.300Z INFO wdio-cli:Launcher: Run onPrepare hook
Starting ChromeDriver 2.45.615355 (d5698f682d8b2742017df6c81e0bd8e6a3063189) on port 9515
Only local connections are allowed.
2019-01-26T23:12:25.226Z INFO wdio-local-runner: Start worker 0-0 with arg: --logLevel,silent,wdio.conf.js

Test Suites:	 0 passed, 1 total (0% completed)
Time:		 🕑  7.89s

 RUNNING  0-0 in chrome - /test/specs/google.test.js

Stdout:
2019-01-26T23:12:22.300Z INFO wdio-cli:Launcher: Run onPrepare hook
Starting ChromeDriver 2.45.615355 (d5698f682d8b2742017df6c81e0bd8e6a3063189) on port 9515
Only local connections are allowed.
2019-01-26T23:12:25.226Z INFO wdio-local-runner: Start worker 0-0 with arg: --logLevel,silent,wdio.conf.js
2019-01-26T23:12:30.109Z DEBUG wdio-local-runner: Runner 0-0 finished with exit code 0

Test Suites:	 0 passed, 1 total (0% completed)
Time:		 🕒  7.90s

"dot" Reporter:
.

"junit" Reporter:
<?xml version="1.0" encoding="UTF-8"?>
<testsuites>
  <testsuite name="webdriver_io_page" timestamp="2019-01-27T00:12:27" time="2.815" tests="1" failures="0" errors="0" skipped="0">
    <properties>
      <property name="specId" value="0"/>
      <property name="suiteName" value="webdriver.io page"/>
      <property name="capabilities" value="chrome"/>
      <property name="file" value="./test/specs/google.test.js"/>
    </properties>
    <testcase classname="chrome.webdriver_io_page" name="should_have_the_right_title" time="2.813"/>
  </testsuite>
</testsuites>

Stdout:
2019-01-26T23:12:22.300Z INFO wdio-cli:Launcher: Run onPrepare hook
Starting ChromeDriver 2.45.615355 (d5698f682d8b2742017df6c81e0bd8e6a3063189) on port 9515
Only local connections are allowed.
2019-01-26T23:12:25.226Z INFO wdio-local-runner: Start worker 0-0 with arg: --logLevel,silent,wdio.conf.js
2019-01-26T23:12:30.109Z DEBUG wdio-local-runner: Runner 0-0 finished with exit code 0
2019-01-26T23:12:30.110Z INFO wdio-cli:Launcher: Run onComplete hook

Test Suites:	 1 passed, 1 total (100% completed)
Time:		 🕓  7.90s
```
</details>

### PR DESCRIPTION
**Pull request by christian-bromann**
## Proposed changes

Our interactive test runner display is not optimal atm as it creates a bunch of noise duplicate logs. This PR is an effort to fix this by replacing it with the `logUpdate` package.

closes #3437

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

## Further comments

work in progress

### CODE REVIEW

File: packages/wdio-config/src/shim.js [hunk 1]
```diff
@@ -60,13 +60,12 @@ export let hasWdioSyncSupport = false
...
     // eslint-disable-next-line import/no-unresolved
     const wdioSync = require('@wdio/sync')
-    log.debug('@wdio/sync found, running tests synchronous')
 
     hasWdioSyncSupport = true
     runFnInFiberContext = wdioSync.runFnInFiberContext
     runTestInFiberContext = wdioSync.runTestInFiberContext
     wrapCommand = wdioSync.wrapCommand
     executeHooksWithArgs = wdioSync.executeHooksWithArgs
 } catch (e) {
```
Comment (abjerstedt) [inline packages/wdio-config/src/shim.js]: You could just remove the error param. I think we're added that Babel plugin

**Comment by mgrybyk** [comment]
@christian-bromann can you please post log output example with and with outputDir?

**Comment by christian-bromann** [comment]
> can you please post log output example with and with outputDir?

@mgrybyk what has `outputDir` to do with the testrunner display?

**Comment by mgrybyk** [comment]
Well, to be short, hell a lot.
If it is set lots of loggers will write data to file, otherwise to console.

**Comment by mgrybyk** [comment]
Anyway it would be good to see some example

**Comment by mgrybyk** [comment]
@christian-bromann are you still working on this one?

**Comment by christian-bromann** [comment]
@mgrybyk yes, was busy in the last weeks and can now pick it up again

**Comment by christian-bromann** [comment]
> Anyway it would be good to see some example

@mgrybyk can you please describe what kind of examples you want to see?

**Comment by christian-bromann** [comment]
Can someone please test this? It is now using [`log-update`](https://www.npmjs.com/package/log-update) which is much better than catching all stdout manually.

This is how it looks like in a normal terminal:
interface

In a CI environment (where `CI` is set to true in the environment) it will be:
interface2

**Comment by mgrybyk** [comment]
I'll be taking a look later on today

**Comment by carlos-gva** [comment]
I'm taking a look right now.

File: packages/wdio-cli/package.json [hunk 2]
```diff
@@ -50,6 +50,7 @@
     "lodash.flattendeep": "^4.4.0",
     "lodash.pickby": "^4.6.0",
     "lodash.union": "^4.6.0",
+    "log-update": "^2.3.0",
```
Comment (carlos-gva) [inline packages/wdio-cli/package.json]: forgot to remove the above `"@wdio/interface": "^5.7.8"`

**Comment by carlos-gva** [comment]
I was testing this using wdio/examples and also my local tests and the logs are much better now. Great look. Many thanks @christian-bromann

**Comment by mgrybyk** [comment]
@christian-bromann tested your code locally and it looks really good!

tried 3 mods:
- in VS Code integrated console (it was good even before)
- iTerm on MacOS ( there were some duplications, now it is good! )
- output to file to emulate jenkins behavior ( there were lots of duplications, looks good now! )

Gist with several examples (with/without outputDir, with/without failures) https://gist.github.com/mgrybyk/ed6b12e5ce8dcdcb5d29789f0f29b1f9

screenshot from iTerm
image

**Comment by mgrybyk** [comment]
Do you know why does `console.warn('console.warn')` is treated as error? Can we change it?
image

File: packages/wdio-cli/src/interface.js [hunk 3]
```diff
@@ -213,95 +255,110 @@ export default class WDIOCLInterface extends EventEmitter {
...
+        this.clearConsole()
         this.printReporters()
+        this.printStdout()
         this.printSummary()
         this.updateClock()
-        this.reset()
+        this.resetClock()
+
+        // eslint-disable-next-line
+        console.log('')
```
Comment (mgrybyk) [inline packages/wdio-cli/src/interface.js]: what does this stand for?

File: packages/wdio-cli/tests/interface.test.js [hunk 4]
```diff
@@ -21,6 +26,7 @@ describe('cli interface', () => {
     })
 
     it('should mark jobs as pass or failed', () => {
+        delete process.env.CI
```
Comment (mgrybyk) [inline packages/wdio-cli/tests/interface.test.js]: may it be moved to before/after each?

File: packages/wdio-local-runner/tests/replQueue.test.js [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    })
+    expect(endFn).toBeCalledTimes(1)
+    expect(startFn2).toBeCalledTimes(1)
+    expect(endFn2).toBeCalledTimes(0)
+
+    await new Promise((resolve) => setTimeout(resolve, 100))
+    expect(queue.isRunning).toBe(false)
+})
+
+// export default class ReplQueue {
```
Comment (mgrybyk) [inline packages/wdio-local-runner/tests/replQueue.test.js]: are you going to remove it?

**Comment by jankcat** [comment]
https://nodejs.org/api/console.html#console_console_warn_data_args
console.warn is an alias for console.error, which writes to stderr

File: packages/wdio-cli/tests/interface.test.js (hunk 4 above)
Comment (christian-bromann) [reply packages/wdio-cli/tests/interface.test.js]: it just affects this one test

File: packages/wdio-cli/src/interface.js (hunk 3 above)
Comment (christian-bromann) [reply packages/wdio-cli/src/interface.js]: it adds a line break at the end of the final report, added a comment

**Comment by christian-bromann** [comment]
@mgrybyk can you have a final look?

**Pull request merged**